### PR TITLE
Improve NextLabelCause Test Coverage

### PR DESCRIPTION
**Pull request by shinigami-777**
Part of [JENKINS-69756](https://issues.jenkins.io/browse/JENKINS-69756)
Added unit tests for `NextLabelCause` class.
Test-coverage before:
nblp1
Test-coverage after:
nblp2

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by yashpal2104** [comment]
Hey @shinigami-777 I think I have written some more tests that would likely increase the coverage to 85%, If we have permission of @MarkEWaite and you would like some help I maybe able to provide with some more tests.

File: src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/NextLabelCauseTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+public class NextLabelCauseTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    /**
+     * Tests whether a job A is able to trigger job B to be executed on a specific node/slave. If it
+     * does not work, the timeout will stop/fail the test after 60 seconds.
+     *
+     * @throws Exception
+     */
```
Comment (MarkEWaite) [inline src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/NextLabelCauseTest.java:22]: This comment is misplaced.  It says that it throws an exception, but the comment will be associated with the declaration of `FreeStyleBuild build`. It needs to either be moved to immediately precede a test that the comment matches or it needs to be deleted.
```suggestion
    /**
     * Tests whether a job A is able to trigger job B to be executed on a specific node/slave. If it
     * does not work, the timeout will stop/fail the test after 60 seconds.
     *
     * @throws Exception
     */
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Thanks very much!  I've pushed a change that switches the test to use equalsverifier instead of testing the equals and hashCode contracts from specific code.  The equalsverifier library is more thorough in testing the equals contract and the hashCode contract.

Could you move the `setup` method contents into the one test that now uses it?  That will further simplify the test.

**Comment by MarkEWaite** [comment]
> Do let me know about your thoughts on this @MarkEWaite and @shinigami-777 , because I am not the author of this PR so I cant change any of the code within this PR I will have to create a new PR for this.

I think that the change to use equalsverifier has increased coverage of the NextLabelCause object to 100%, at least according to the "indirect changes" report on ci.jenkins.io.  I don't think that the NextLabelCause object will need any more automated tests once this pull request has merged.

**Comment by shinigami-777** [comment]
> Thanks very much! I've pushed a change that switches the test to use equalsverifier instead of testing the equals and hashCode contracts from specific code. The equalsverifier library is more thorough in testing the equals contract and the hashCode contract.
> 
> Could you move the `setup` method contents into the one test that now uses it? That will further simplify the test.
Yes. I have moved it.

**Review by MarkEWaite — APPROVED** [review]
Thanks

File: src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/NextLabelCauseTest.java (hunk 1 above)
Comment (shinigami-777) [reply src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/NextLabelCauseTest.java:21]: I tried removing the comment and it led to build failure. Keeping it to immediately preceding the `testGetShortDescription` test.

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Pull request merged**